src/charm.py

```diff
@@ -179,6 +182,29 @@ def _on_get_initial_password_action(self, event):
             logger.error("Action get-initial-password failed. Replica consensus not exists")
             event.fail("Default admin password has not been generated yet.")
 
+    def _on_rotate_wordpress_secrets_action(self, event):
+        """Handle the rotate-wordpress-secrets action."""
+        if self._replica_consensus_reached():
+            if self.unit.is_leader():
+                # Update the secrets in peer relation.
+                replica_relation_data = self._replica_relation_data()
+                secrets = self._generate_wp_secret_keys()
```

Comment:
It looks like this method will also rotate the password salts which as a consequence will lead to all existing passwords behind unusable for users.
Is it really expected ? It could totally break the wordpress

Reply:
I confirmed with @weiiwang01 that this is the intended behavior.
This is the same as the original PR on launchpad this PR is based on.
https://code.launchpad.net/~jsimpso/charm-k8s-wordpress/+git/charm-k8s-wordpress/+merge/419738

Reply:
I think these are just the salt for cookies, so updating these will not reset user's password.

Here's some explanation from WordPress:
```php

/**#@+
 * Authentication unique keys and salts.
 *
 * Change these to different unique phrases! You can generate these using
 * the {@link https://api.wordpress.org/secret-key/1.1/salt/ WordPress.org secret-key service}.
 *
 * You can change these at any point in time to invalidate all existing cookies.
 * This will force all users to have to log in again.
 *
 * @since 2.6.0
 */
define( 'AUTH_KEY',         'put your unique phrase here' );
define( 'SECURE_AUTH_KEY',  'put your unique phrase here' );
define( 'LOGGED_IN_KEY',    'put your unique phrase here' );
define( 'NONCE_KEY',        'put your unique phrase here' );
define( 'AUTH_SALT',        'put your unique phrase here' );
define( 'SECURE_AUTH_SALT', 'put your unique phrase here' );
define( 'LOGGED_IN_SALT',   'put your unique phrase here' );
define( 'NONCE_SALT',       'put your unique phrase here' );

```